### PR TITLE
Add an ability to disable container for the field.

### PR DESCRIPTION
**Pull request by armpogart**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ✔️
| Fixed issues  | Fixes one of the points (7) mentioned in [#17](https://github.com/yiisoft/form/issues/17#issuecomment-672885901)

### CODE REVIEW

File: src/Widget/Field.php [hunk 1]
```diff
@@ -741,17 +747,19 @@ private function addInputId(): string
...
      * @param string $attribute Form model property this widget is rendered for.
-     * @param array $options The HTML attributes for the widget container tag.
+     * @param array $options The HTML attributes for the widget container tag if it is enabled.
      * See {@see \Yiisoft\Html\Html::renderTagAttributes()} for details on how attributes are being rendered.
+     * @param bool $containerEnabled whether the widget container is enabled. Default: true
      *
      * @return self
      */
-    public function config(FormModelInterface $data, string $attribute, array $options = []): self
+    public function config(FormModelInterface $data, string $attribute, array $options = [], $containerEnabled = true): self
```
Comment (terabytesoftw) [inline src/Widget/Field.php:756]: I think a containerEnabled() method would be better.

Comment (armpogart) [reply src/Widget/Field.php:756]: @terabytesoftw I first though also about adding a method, but it would be weird to have an ability to do `containerEnabled(true)` if it is already enabled by default. It would make sense to have a separate method if we also `$options` there too. In that case we will remove `$options` from `config` method and add `container` method with 2 arguments `function container($enabled = true, array $options = [])`. What do you think?

Comment (armpogart) [reply src/Widget/Field.php:756]: If `$options` for the container are configured in `config()` method and enabling/disabling it is a separate method, it becomes weird to have an ability to set the options in config and then disable the container. So I think it is better to keep these 2 things together either in `config` method as it is now or a separate method with both of them.

Comment (terabytesoftw) [reply src/Widget/Field.php:756]: We can change the name to `enclosedByContainer()` and in the same way pass the options for the container.

File: src/Widget/Field.php [hunk 2]
```diff
@@ -737,21 +744,34 @@ private function addInputId(): string
...
+     *
+     * @param bool $containerEnabled enabled/disable container for the widget.
+     * @param array $options The HTML attributes for the widget container tag if it is enabled.
+     * See {@see \Yiisoft\Html\Html::renderTagAttributes()} for details on how attributes are being rendered.
+     * @return self
+     */
+    public function enclosedByContainer($containerEnabled = true, array $options = []): self
+    {
+        $new = clone $this;
         $new->options = $options;
```
Comment (xepozz) [inline src/Widget/Field.php:773]: Why did you move `$options` here?

File: src/Widget/Field.php [hunk 3]
```diff
@@ -737,21 +744,34 @@ private function addInputId(): string
...
+
+    /**
+     * Generate a container tag for {@see attribute}.
+     *
+     * @param bool $containerEnabled enabled/disable container for the widget.
+     * @param array $options The HTML attributes for the widget container tag if it is enabled.
+     * See {@see \Yiisoft\Html\Html::renderTagAttributes()} for details on how attributes are being rendered.
+     * @return self
+     */
+    public function enclosedByContainer($containerEnabled = true, array $options = []): self
```
Comment (xepozz) [inline src/Widget/Field.php:770]:
```suggestion
    public function enclosedByContainer(bool $containerEnabled, array $options = []): self
```

File: src/Widget/Field.php (hunk 2 above)
Comment (armpogart) [reply src/Widget/Field.php:773]: See [discussion](https://github.com/yiisoft/form/pull/33#discussion_r469294033) on @terabytesoftw requested changes above.

Comment (xepozz) [reply src/Widget/Field.php:773]: What about to split these onto 2 independent methods?

Comment (armpogart) [reply src/Widget/Field.php:773]: Do you mean something like `containerEnabled(bool $enabled)` and `containerOptions(array $options = [])`?
I have tried to replicate `label(bool $enabledLabel = true, array $options = [], ?string $label = null)` method, which is already there and combines enabling/disabling label with its' options.

Comment (armpogart) [reply src/Widget/Field.php:773]: I'm ok with any approach allowing to disable container, but I would like to keep behaviors persistent. So in case of 2 independent methods, the same thing must be done for label.

Comment (xepozz) [reply src/Widget/Field.php:773]: Ok let it be. We can think about it later.

Comment (samdark) [reply src/Widget/Field.php:773]: It can be split to withoutContainer() and withContainer(array $options).

Comment (armpogart) [reply src/Widget/Field.php:773]: @samdark What about label in that case?

**Comment by terabytesoftw** [comment]
👍

File: src/Widget/Field.php (hunk 2 above)
Comment (samdark) [reply src/Widget/Field.php:773]: That's just a method split nothing is changed about label.

Comment (armpogart) [reply src/Widget/Field.php:773]: @samdark I mean the same must be done with `label` method (which works the same now) to be consistent.
